### ISSUE TITLE
Parsing issues with non MSVC toolchain + solution but a custom workspace parser settings instead

### ISSUE THREAD

**Issue by Scr3amer** [slynch8/10x#2417]
Hello there !

I was trying to use 10x in a custom workspace, I setup the parser include folders based on the output of my clang/llvm/mingw/ucrt toolchain include list but it has a hard time parsing properly the standard headers.

For instance when I include <cstdio> it doesn't find puts and other functions.
I need to include <stdio.h> for it to work somehow.

I wonder if this is due to a lack of support of preprocessor directives such as #include_next maybe ?
Or maybe I need to put all the default defines myself of that toolchain ? __clang__ and such ?

Cheers,
Scr3am

**Comment by slynch8** [comment]
Hello.

The parser will only parse files in the workspace or external files that are included. As far as I'm aware you do need to include stdio.h to use puts otherwise it won't compile.

I've never come across #include_next before. I'm not sure how it would solve this problem though?

One other thing that might help is adding force includes. This adds includes to the top of all cpp files. You can find it in the workspace settings.

**Comment by Scr3amer** [comment]
Sorry I think I was not really clear in my initial message. Here are some additional information and rephrasing:

> As far as I'm aware you do need to include stdio.h to use puts otherwise it won't compile
I am writing c++ code so I do include cstdio which inside includes stdio.h. However because I am not using MSVC rather a custom toolchain (a MinGW based one) headers are different and it seems the parser of 10x doesn't support them.

> One other thing that might help is adding force includes. 
I found that option but I do not think that would fix a header parsing issue.

The toolchain I am using is this one if you wanna see how the headers look like : https://github.com/mstorsjo/llvm-mingw

Here is some content of mstorsjo 16.0.6 toolchain file include\c++\v1\stdio.h

![image](https://github.com/slynch8/10x/assets/9144188/d3202553-99d5-41b7-ad4e-d2d19e365d8a)

![image](https://github.com/slynch8/10x/assets/9144188/337bef0d-9347-4036-80ee-1da3e58bc8cd)

**Comment by slynch8** [comment]
I see, so it does need #include_next support

I wonder, in this case, will it work if I just make include_next work the same as include?

**Comment by slynch8** [comment]
Looking at it more, it looks like it won't work. I will need to support it properly

**Comment by Scr3amer** [comment]
> Looking at it more, it looks like it won't work. I will need to support it properly

Exactly, as far as I know, include_next goes in the chain of include folders and tries to find a file with that name but later in that chain. This also explains why the order of the include folders matters.
